contrastive_vi/module/total_contrastive_vi.py

```diff
@@ -703,20 +710,30 @@ def loss(
         loss = torch.mean(
             reconst_loss_gene
             + reconst_loss_protein
-            + kl_z
-            + kl_s
-            + kl_library_gene
-            + kl_div_back_pro
+            + kl_weight * kl_div_z
+            + kl_weight * kl_div_s
+            + kl_weight * self.wasserstein_penalty*wasserstein_loss
+            + kl_div_l_gene
+            + kl_weight * kl_div_back_pro
         )
 
+        reconst_losses = dict(
+            reconst_loss_gene=reconst_loss_gene,
+            reconst_loss_protein=reconst_loss_protein,
+        )
         kl_local = dict(
-            kl_z=kl_z,
-            kl_s=kl_s,
-            kl_library_gene=kl_library_gene,
+            kl_div_z=kl_div_z,
+            kl_div_l_gene=kl_div_l_gene,
             kl_div_back_pro=kl_div_back_pro,
         )
         kl_global = torch.tensor(0.0)
-        return LossRecorder(loss, reconst_losses, kl_local, kl_global)
+        return LossRecorder(
+            loss,
+            reconst_losses,
+            kl_local,
+            kl_global=kl_global,
+            wasserstein_loss=torch.sum(wasserstein_loss)
```

Comment:
Again, maybe the mean should be taken here to be consistent with the overall loss.

Reply:
See previous comment.